### ISSUE TITLE
Question about redux == "swap"

### ISSUE THREAD

**Issue by ylsung** [google-research/remixmatch#7]
Thank you for the codes. 

When reading the codes, I wonder what the purpose of swapping the logits in `guess_label` is.

https://github.com/google-research/remixmatch/blob/f7061ebf055227cbeb5c6fced1ce054e0ceecfcd/remixmatch_no_cta.py#L38-L39

Also, the paper said that only the guessed labels of the weakly augmented data are used in training. However, in `guess_label`, it seems that labels of all the augmented data are used if `redux == "swap"`(default setting). Please let me know if I misunderstand.

Thank you in advance.

**Comment by carlini** [comment]
When experimenting with ReMixMatch, we tried several different ways of computing the guessed label. This included one idea where we swapped the predictions of two augmented unlabeled examples. Example u1 would target the guessed label of u2, and example u2 would target the guessed label of u1.

This idea ultimately worked less well than what we do in the paper, and make every strongly augmented image target the weakly augmented image. However, we left in the option to experiment with different modes.
